test/mock/src/mockEnvironment.ts

```diff
@@ -112,14 +114,29 @@ export function doInstall(options: MockOptionsJSON) {
             error = helpers.Errors.InternalError;
           }
           const response = { id: msg.id, error };
-          setImmediate(() => receiveMessageCallback({ data: JSON.stringify(response) }));
+          setImmediate(() => receiveDataCallback(JSON.stringify(response)));
         }
       );
     },
   };
+
+  // Expose mock environment to Replay devtools app.
+  window.__mockEnvironmentForTesting = mockEnvironment;
+
+  // Configure the protocol to use our mock WebSocket message method.
+  const { flushQueuedMessages, responseDataHandler } = setCustomSendMessageHandler(
+    mockEnvironment.sendSocketMessage
+  );
+
+  // Configure the protocol's default socket message handler to receive messages from the mocked WebSocket.
+  mockEnvironment.setSocketDataHandler(responseDataHandler);
+
+  // Now that the mock environment has been configured, flush any pending messages.
+  flushQueuedMessages();
```

Comment:
Ah, I don't think any of these lines of code can run from within the context of `page.evaluate` 😢  Need to re-think.